google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/ModackRequestData.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.common.collect.Lists;
+import java.util.*;
+
+class ModackRequestData {
+  private final int deadlineExtensionSeconds;
+  private List<AckRequestData> ackRequestData;
+
+  ModackRequestData(int deadlineExtensionSeconds) {
+    this.deadlineExtensionSeconds = deadlineExtensionSeconds;
+    this.ackRequestData = new ArrayList<AckRequestData>();
+  }
+
+  ModackRequestData(int deadlineExtensionSeconds, AckRequestData... ackRequestData) {
+    this.deadlineExtensionSeconds = deadlineExtensionSeconds;
+    this.ackRequestData = Arrays.asList(ackRequestData);
+  }
+
+  ModackRequestData(int deadlineExtensionSeconds, List<AckRequestData> ackRequestData) {
+    this.deadlineExtensionSeconds = deadlineExtensionSeconds;
+    this.ackRequestData = ackRequestData;
+  }
+
+  public int getDeadlineExtensionSeconds() {
+    return deadlineExtensionSeconds;
+  }
+
+  public List<AckRequestData> getAckIdMessageFutures() {
+    return ackRequestData;
+  }
+
+  public ModackRequestData addAckIdMessageFuture(AckRequestData ackRequestData) {
+    this.ackRequestData.add(ackRequestData);
+    return this;
+  }
+
+  public ModackRequestData addAllAckIdMessageFuture(List<AckRequestData> ackRequestData) {
+    this.ackRequestData.addAll(ackRequestData);
+    return this;
+  }
+
+  public List<ModackRequestData> partitionByAckId(int batchSize) {
+    // Helper function to return a new list of ModackWithMessageFutures with a batchSize number of
+    // ackIds
+    List<ModackRequestData> modackRequestData = new ArrayList<ModackRequestData>();
+    for (List<AckRequestData> ackRequestData : Lists.partition(this.ackRequestData, batchSize)) {
+      modackRequestData.add(new ModackRequestData(this.deadlineExtensionSeconds, ackRequestData));
+    }
+    return modackRequestData;
+  }
+
+  public static List<ModackRequestData> partitionByAckId(
```

Comment:
I don't think we really even need this helper, as it is only used in one place and requires unnecessary extra work.

When used, we basically:

Loop through each ModackRequestData, and partition the ackRequestData. Then, we loop through the partitioned data, and have to create a new ModackRequestData object that is added to the returned list. When the list is returned, we again loop through the returned list to send each modack.

If this function wasn't used, we would just have to:

Loop through each ModackRequestData and partition the data. Loop through the partitioned data and send the modack using partitioned data. This is much simpler.

Reply:
Updated.